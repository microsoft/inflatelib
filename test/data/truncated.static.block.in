
# This generates a file with a single compressed block with no "end of block" symbol
# NOTE: This test is valid for both Deflate and Deflate64
>1
1       # bfinal = true
01      # Compressed with static codes

# This writes the lyrics to "Three Blind Mice"
>>1
10000100 10011000 10100010 10010101 10010101 01010000 10010010 10011100 10011001 10011110 10010100 01010000 10011101 10011001 10010011 10010101 01011110 00111010
0001100 >1 1 >>1 01000 >1 001 >>1
10000011 10010101 10010101 01010000 10011000 10011111 10100111 01010000 10100100 10011000 10010101 10101001 01010000 10100010 10100101 10011110 01011110 00111010
0001100 >1 1 >>1 01000 >1 001 >>1
10000100 10011000 10010101 10101001 01010000 10010001 10011100 10011100 01010000 10100010 10010001 10011110 01010000 10010001 10010110 10100100 10010101 10100010 01010000 10100100 10011000 10010101 01010000 10010110 10010001 10100010 10011101 10010101 10100010 01010111 10100011 01010000 10100111 10011001 10010110 10010101 01011100 00111010
10000111 10011000 10011111 01010000 10010011 10100101 10100100 01010000 10011111 10010110 10010110 01010000 10100100 10011000 10010101 10011001 10100010 01010000 10100100 10010001 10011001 10011100 10100011 01010000 10100111 10011001 10100100 10011000 01010000 10010001 01010000 10010011 10010001 10100010 10100110 10011001 10011110 10010111 01010000 10011011 10011110 10011001 10010110 10010101 01011110 00111010
01110100 10011001 10010100 01010000 10101001 10011111 10100101 01010000 10010101 10100110 10010101 10100010 01010000 10100011 10010101 10010101 01010000 10100011 10100101 10010011 10011000 01010000 10010001 01010000 10100011 10011001 10010111 10011000 10100100 01010000 10011001 10011110 01010000 10101001 10011111 10100101 10100010 01010000 10011100 10011001 10010110 10010101 00111010
01110001 10100011 01010000 10100100 0001011 >1 0 >>1 01110 >1 110111 >>1 01101111 00111010

# The remainder of this data was originally intended to repeat all of the text once, however the repeat has been truncated
11000011 >1 11000 >>1
# At this point in time, there are five bits left in the last byte. Leaving them as zero is problematic as that's a valid
# distane code (distance = 1, no extra bits). Fill up those last 5 bits; this will still be truncated as we're still
# expecting an additional 5 extra bits
01111 # >1 011010 >>1
# 0000000
