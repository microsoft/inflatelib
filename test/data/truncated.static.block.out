
"Three blind mice.\n"
"Three blind mice.\n"
"See how they run.\n"
"See how they run.\n"
"They all ran after the farmer's wife,\n"
"Who cut off their tails with a carving knife.\n"
"Did you ever see such a sight in your life\n"
"As three blind mice?\n"
